code/parse/sexp/LuaAISEXP.cpp

```diff
@@ -0,0 +1,150 @@
+//
+//
+#include "LuaAISEXP.h"
+
+#include "localization/localize.h"
+#include "parse/parselo.h"
+#include "parse/sexp/LuaSEXP.h"
+
+using namespace luacpp;
+
+namespace sexp {
+static const SCP_unordered_set<int> allowed_oswpt_parameters{ OPF_SHIP, OPF_WING, OPF_SHIP_POINT, OPF_SHIP_WING, OPF_SHIP_WING_WHOLETEAM, OPF_SHIP_WING_SHIPONTEAM_POINT, OPF_SHIP_WING_POINT, OPF_SHIP_WING_POINT_OR_NONE };
+
+LuaAISEXP::LuaAISEXP(const SCP_string& name) : DynamicSEXP(name) {
+}
+
+void LuaAISEXP::initialize() {
};
+
+int LuaAISEXP::getMinimumArguments() {
	return _arg_type == -1 ? 1 : 2;
};
+
+int LuaAISEXP::getMaximumArguments() {
	return getMinimumArguments();
};
+
+int LuaAISEXP::getArgumentType(int argnum) const {
	return argnum == 0 && _arg_type != -1 ? _arg_type : OPF_POSITIVE;
};
+
+int LuaAISEXP::execute(int /*node*/) {
+	UNREACHABLE("Tried to execute AI Lua SEXP %s! AI-Goal SEXPs should never be run.", _name.c_str());
+	return SEXP_CANT_EVAL;
+}
+
+int LuaAISEXP::getReturnType() {
	return OPR_AI_GOAL;
};
+
+int LuaAISEXP::getSubcategory() {
	return -1;
};
+
+int LuaAISEXP::getCategory() {
	return OP_CATEGORY_AI;
};
+
+void LuaAISEXP::parseTable() {
+	SCP_stringstream help_text;
+	help_text << _name << "\r\n";
+
+	required_string("$Description:");
+
+	SCP_string description;
+	stuff_string(description, F_NAME);
+
+	help_text << "\t" << description << "\r\n";

```

Comment:
If you want to allow multi-line descriptions, it will be necessary to add the `parseCheckEndOfDescription` logic from LuaSEXP.